andela/ah-rambo-backend
# #167190528 Build Email Verification Service

#### What does this PR do?
- It adds the email verification function and checks to prevent unverified users after some time

#### Description of Task proposed in this pull request?
- Create an email module for sending any kind of email to users
- Create a template for email verification service
- Send verification emails when the user signs up
- Add route and controller to verify emails
- Prevent unverified users from access to the app after 24 hours of registration

#### How should this be manually tested (Quality Assurance)?
- Register a new account by making a `POST` request to the `/users/create/` endpoint. You will receive an email. Click the link in the email to verify your email.

#### What are the relevant pivotal tracker stories?
[[#167190528]](https://www.pivotaltracker.com/story/show/167190528)

#### Any background context you want to add (Operations Impact)?
- This PR adds email verification functionality. It allows the app to send and verify the emails of users after they have signed up. It also prevents users who haven't been verified from using the app.

#### What I have learned working on this feature: 
- I have learned how to create reusable email functions
- I have learned how to stub external libraries in testing

#### Screenshots: 
<img width="1440" alt="Screenshot 2019-08-07 at 8 57 42 PM" src="https://user-images.githubusercontent.com/42539315/62653754-26c0b400-b956-11e9-8669-c350d92f0863.png">
<img width="1440" alt="Screenshot 2019-08-07 at 8 58 01 PM" src="https://user-images.githubusercontent.com/42539315/62653772-317b4900-b956-11e9-8662-a0616e0e09e3.png">
